homeassistant/components/emoncms_history/__init__.py

```diff
@@ -42,61 +43,50 @@
 )
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the Emoncms history component."""
-    conf = config[DOMAIN]
-    whitelist = conf.get(CONF_WHITELIST)
-
-    def send_data(url, apikey, node, payload):
-        """Send payload data to Emoncms."""
+async def async_send_to_emoncms(
+    hass: HomeAssistant,
+    emoncms_client: EmoncmsClient,
+    whitelist: list[str],
+    node: str | int,
+    _: datetime,
+) -> None:
+    """Send data to Emoncms."""
+    payload_dict = {}
+
+    for entity_id in whitelist:
+        state = hass.states.get(entity_id)
+        if state is None or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE):
+            continue
         try:
-            fullurl = f"{url}/input/post.json"
-            data = {"apikey": apikey, "data": payload}
-            parameters = {"node": node}
-            req = requests.post(
-                fullurl, params=parameters, data=data, allow_redirects=True, timeout=5
-            )
-
-        except requests.exceptions.RequestException:
-            _LOGGER.error("Error saving data '%s' to '%s'", payload, fullurl)
-
-        else:
-            if req.status_code != HTTPStatus.OK:
-                _LOGGER.error(
-                    "Error saving data %s to %s (http status code = %d)",
-                    payload,
-                    fullurl,
-                    req.status_code,
-                )
-
-    def update_emoncms(time):
-        """Send whitelisted entities states regularly to Emoncms."""
-        payload_dict = {}
-
-        for entity_id in whitelist:
-            state = hass.states.get(entity_id)
+            payload_dict[entity_id] = state_helper.state_as_number(state)
+        except ValueError:
+            continue
 
-            if state is None or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE):
-                continue
-
-            try:
-                payload_dict[entity_id] = state_helper.state_as_number(state)
-            except ValueError:
-                continue
-
-        if payload_dict:
-            payload = ",".join(f"{key}:{val}" for key, val in payload_dict.items())
+    if payload_dict:
+        try:
+            await emoncms_client.async_input_post(data=payload_dict, node=node)
+            _LOGGER.debug("Sent data to Emoncms: %s", payload_dict)
```

Comment:
Only have things in the try block that can raise

Reply:
done, thanks